candlex/lib/candlex/backend.ex

```diff
@@ -0,0 +1,423 @@
+defmodule Candlex.Backend do
```

Comment:
This is looking nice. But I'm going to hold off on further reviews for now.

I'm not sure if we should add this to the Nx repo like it is now.